From 59f7098bfbd63f9b3c19d7d297a8f90870a1893d Mon Sep 17 00:00:00 2001
From: Joe Slater <joe.slater@windriver.com>
Date: Mon, 31 May 2021 14:50:38 +0800
Subject: [PATCH] samba: defeat iconv test

A test was added when configuring samba which requires target code
to be executed.  In general, this will not work, so we eliminate it.

Upstream-Status: Pending

Signed-off-by: Joe Slater <joe.slater@windriver.com>
---
 lib/util/charset/wscript_configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/util/charset/wscript_configure b/lib/util/charset/wscript_configure
index 9c27fc6..845eb36 100644
--- a/lib/util/charset/wscript_configure
+++ b/lib/util/charset/wscript_configure
@@ -17,6 +17,8 @@ if (conf.CHECK_FUNCS_IN('iconv_open', 'iconv', checklibc=False, headers='iconv.h
     
     conf.DEFINE('HAVE_NATIVE_ICONV', 1)
 
+"""
+[added for 4.7.0, but breaks cross-compilation]
 conf.CHECK_CODE('''
                 uint8_t inbuf[2] = { 0x30, 0xdf };
                 uint8_t outbuf[4] = { 0 };
@@ -36,6 +38,7 @@ conf.CHECK_CODE('''
                 msg='Checking errno of iconv for illegal multibyte sequence',
                 lib='iconv',
                 headers='errno.h iconv.h')
+"""
 
 if conf.CHECK_CFG(package='icu-i18n',
                args='--cflags --libs',
-- 
2.17.1

